test/e2e/conftest.py

```diff
@@ -1,53 +1,250 @@
+import asyncio
+import configparser
 import logging
 import os
 import uuid
 from contextlib import contextmanager
 from pathlib import Path
-from typing import Any, Callable, Iterator
+from typing import Any, Callable, Iterator, Optional, Tuple
 
 import ddtrace
 import pytest
+import xdist
+from filelock import FileLock
 
 import layer
 from layer.clients.layer import LayerClient
-from layer.config import ClientConfig, Config, ConfigManager
+from layer.config import DEFAULT_LAYER_PATH, ClientConfig, Config, ConfigManager
+from layer.contracts.accounts import Account
 from layer.contracts.fabrics import Fabric
 from layer.contracts.projects import Project
-from layer.projects.project_runner import ProjectRunner
+from layer.exceptions.exceptions import LayerClientResourceNotFoundException
 from test.e2e.assertion_utils import E2ETestAsserter
 
 
 logger = logging.getLogger(__name__)
 
-TEST_ASSETS_PATH = Path(__file__).parent / "assets"
-TEST_TEMPLATES_PATH = TEST_ASSETS_PATH / "templates"
-
-REFERENCE_E2E_PROJECT_NAME = "e2e-reference-project-titanic"
-REFERENCE_E2E_PROJECT_PYTEST_CACHE_KEY = "e2e_reference_project_id"
+TEST_SESSION_CONFIG_TMP_PATH = DEFAULT_LAYER_PATH / "e2e_test_session.ini"
 
 
 def pytest_sessionstart(session):
+    """
+    We use this hook to create a single new organization account to be used in all
+    organization e2e tests.
+
+    We clean it up (teardown) using `_track_session_counts_and_cleanup` fixture.
+    """
     if os.getenv("CI"):
         ddtrace.patch_all()
 
+    if xdist.is_xdist_worker(session):
+        return
+
+    org_account = _read_organization_account_from_test_session_config()
+    if org_account:
+        raise Exception(
+            f"pytest_sessionstart test session config already setup with account {org_account.name} {org_account.id}"
+        )
```

Comment:
If I get in this scenario, how do I unblock myself? Would it be better if we just try to delete it and move on? Maybe error if we fail?

Reply:
this was to prevent a bad setup and should fail early. but i should delete then before failing so we don't leak.
i would still have this fail early instead of silently ignore this illegal state